### PR TITLE
UHF-10581 

### PR DESCRIPTION
**Pull request by rpnykanen**
Added test features

## How to install

* Make sure your instance is up and running on correct branch.
  * `git checkout UHF-10581`
  * `make fresh`

## How to test

- Create new job listing, set  publishing time from sidebar and save as unpublished
- After the scheduled time, run drush cron
- Run drush sqlc and `select * from queue`, you should see the indexing job in the table

### CODE REVIEW

File: public/modules/custom/helfi_google_api/src/Plugin/QueueWorker/IndexingWorker.php [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+      $container->get('entity_type.manager'),
+      $container->get('Drupal\helfi_google_api\JobIndexingService'),
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function processItem($data): void {
+    if (!$data['nid']) {
```
Comment (tuutti) [inline public/modules/custom/helfi_google_api/src/Plugin/QueueWorker/IndexingWorker.php:66]: This should probably be `if isset` check.

File: public/modules/custom/helfi_google_api/src/Plugin/QueueWorker/IndexingWorker.php [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+
+    if (!$node instanceof JobListing) {
+      return;
+    }
+
+    try {
+      $this->jobIndexingService->indexEntity($node);
+    }
+    catch (\Exception $e) {
+      // Handled in service.
```
Comment (tuutti) [inline public/modules/custom/helfi_google_api/src/Plugin/QueueWorker/IndexingWorker.php:81]: Should this be re-queued if it fails? If so, then you should throw `RequeueException`

File: public/modules/custom/helfi_google_api/tests/src/Kernel/IndexingTest.php [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+use Drupal\redirect\Entity\Redirect;
+use Drupal\Tests\helfi_api_base\Functional\ExistingSiteTestBase;
+use Prophecy\PhpUnit\ProphecyTrait;
+
+/**
+ * Tests file name transliteration.
+ *
+ * @group helfi_google_api
+ */
+class IndexingTest extends ExistingSiteTestBase {
```
Comment (tuutti) [inline public/modules/custom/helfi_google_api/tests/src/Kernel/IndexingTest.php:20]: You should test what happens if the API request fails as well.

**Review by tuutti — CHANGES_REQUESTED** [review]

File: public/modules/custom/helfi_google_api/src/Plugin/QueueWorker/IndexingWorker.php (hunk 2 above)
Comment (rpnykanen) [reply public/modules/custom/helfi_google_api/src/Plugin/QueueWorker/IndexingWorker.php]: exception only happens in guzzle error so yes

**Review by tuutti — APPROVED** [review]

**Pull request merged**